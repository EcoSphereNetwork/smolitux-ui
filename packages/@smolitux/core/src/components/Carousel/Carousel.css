.smolitux-carousel {
  position: relative;
  width: 100%;
  overflow: hidden;
}

/* Aspect Ratio Container */
.smolitux-carousel-container {
  position: relative;
  width: 100%;
}

.smolitux-carousel-container--1-1 {
  padding-top: 100%;
}

.smolitux-carousel-container--4-3 {
  padding-top: 75%;
}

.smolitux-carousel-container--16-9 {
  padding-top: 56.25%;
}

.smolitux-carousel-container--21-9 {
  padding-top: 42.85%;
}

/* Slides */
.smolitux-carousel-slides {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  display: flex;
  transition: transform 0.3s ease-in-out;
}

.smolitux-carousel-slide {
  flex: 0 0 100%;
  width: 100%;
  height: 100%;
  position: relative;
}

/* Animation: Slide */
.smolitux-carousel--animation-slide .smolitux-carousel-slides {
  display: flex;
}

/* Animation: Fade */
.smolitux-carousel--animation-fade .smolitux-carousel-slides {
  display: block;
}

.smolitux-carousel--animation-fade .smolitux-carousel-slide {
  position: absolute;
  top: 0;
  left: 0;
  opacity: 0;
  transition: opacity 0.3s ease-in-out;
  pointer-events: none;
}

.smolitux-carousel--animation-fade .smolitux-carousel-slide--active {
  opacity: 1;
  pointer-events: auto;
}

/* Animation: None */
.smolitux-carousel--animation-none .smolitux-carousel-slides {
  transition: none;
}

.smolitux-carousel--animation-none .smolitux-carousel-slide {
  display: none;
}

.smolitux-carousel--animation-none .smolitux-carousel-slide--active {
  display: block;
}

/* Navigation Arrows */
.smolitux-carousel-arrow {
  position: absolute;
  top: 50%;
  transform: translateY(-50%);
  width: 2.5rem;
  height: 2.5rem;
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: var(--carousel-arrow-bg, rgba(255, 255, 255, 0.8));
  color: var(--carousel-arrow-color, #1f2937);
  border-radius: 50%;
  cursor: pointer;
  z-index: 10;
  transition: all 0.2s ease-in-out;
  border: none;
  outline: none;
}

.smolitux-carousel-arrow:hover {
  background-color: var(--carousel-arrow-hover-bg, rgba(255, 255, 255, 0.9));
}

.smolitux-carousel-arrow:focus {
  box-shadow: 0 0 0 3px rgba(59, 130, 246, 0.5);
}

.smolitux-carousel-arrow--prev {
  left: 1rem;
}

.smolitux-carousel-arrow--next {
  right: 1rem;
}

.smolitux-carousel-arrow--disabled {
  opacity: 0.5;
  cursor: not-allowed;
}

.smolitux-carousel-arrow-icon {
  width: 1.5rem;
  height: 1.5rem;
}

/* Indicators */
.smolitux-carousel-indicators {
  position: absolute;
  bottom: 1rem;
  left: 0;
  right: 0;
  display: flex;
  justify-content: center;
  gap: 0.5rem;
  z-index: 10;
}

.smolitux-carousel-indicator {
  width: 0.75rem;
  height: 0.75rem;
  border-radius: 50%;
  background-color: var(--carousel-indicator-bg, rgba(255, 255, 255, 0.5));
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  border: none;
  outline: none;
  padding: 0;
}

.smolitux-carousel-indicator:hover {
  background-color: var(--carousel-indicator-hover-bg, rgba(255, 255, 255, 0.8));
}

.smolitux-carousel-indicator:focus {
  box-shadow: 0 0 0 3px rgba(59, 130, 246, 0.5);
}

.smolitux-carousel-indicator--active {
  background-color: var(--carousel-indicator-active-bg, white);
  transform: scale(1.2);
}

/* Thumbnail Indicators */
.smolitux-carousel-thumbnails {
  position: relative;
  display: flex;
  justify-content: center;
  gap: 0.5rem;
  margin-top: 0.5rem;
  overflow-x: auto;
  padding: 0.5rem 0;
}

.smolitux-carousel-thumbnail {
  width: 4rem;
  height: 3rem;
  border-radius: 0.25rem;
  overflow: hidden;
  cursor: pointer;
  transition: all 0.2s ease-in-out;
  border: 2px solid transparent;
  padding: 0;
  background: none;
}

.smolitux-carousel-thumbnail:hover {
  opacity: 0.9;
}

.smolitux-carousel-thumbnail:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(59, 130, 246, 0.5);
}

.smolitux-carousel-thumbnail--active {
  border-color: var(--carousel-thumbnail-active-border-color, #3b82f6);
}

.smolitux-carousel-thumbnail img {
  width: 100%;
  height: 100%;
  object-fit: cover;
}

/* Disabled State */
.smolitux-carousel--disabled {
  opacity: 0.7;
  pointer-events: none;
}

/* Autoplay Progress */
.smolitux-carousel-progress {
  position: absolute;
  bottom: 0;
  left: 0;
  height: 3px;
  background-color: var(--carousel-progress-bg, rgba(59, 130, 246, 0.5));
  transition: width 0.1s linear;
  z-index: 10;
}

/* Responsive Adjustments */
@media (max-width: 640px) {
  .smolitux-carousel-arrow {
    width: 2rem;
    height: 2rem;
  }
  
  .smolitux-carousel-arrow-icon {
    width: 1.25rem;
    height: 1.25rem;
  }
  
  .smolitux-carousel-arrow--prev {
    left: 0.5rem;
  }
  
  .smolitux-carousel-arrow--next {
    right: 0.5rem;
  }
  
  .smolitux-carousel-indicators {
    bottom: 0.5rem;
  }
  
  .smolitux-carousel-indicator {
    width: 0.5rem;
    height: 0.5rem;
  }
  
  .smolitux-carousel-thumbnail {
    width: 3rem;
    height: 2.25rem;
  }
}

/* Dark Mode */
.dark .smolitux-carousel-arrow {
  background-color: var(--carousel-arrow-bg-dark, rgba(31, 41, 55, 0.8));
  color: var(--carousel-arrow-color-dark, #f9fafb);
}

.dark .smolitux-carousel-arrow:hover {
  background-color: var(--carousel-arrow-hover-bg-dark, rgba(31, 41, 55, 0.9));
}

.dark .smolitux-carousel-indicator {
  background-color: var(--carousel-indicator-bg-dark, rgba(75, 85, 99, 0.5));
}

.dark .smolitux-carousel-indicator:hover {
  background-color: var(--carousel-indicator-hover-bg-dark, rgba(75, 85, 99, 0.8));
}

.dark .smolitux-carousel-indicator--active {
  background-color: var(--carousel-indicator-active-bg-dark, #9ca3af);
}

.dark .smolitux-carousel-thumbnail--active {
  border-color: var(--carousel-thumbnail-active-border-color-dark, #60a5fa);
}

.dark .smolitux-carousel-progress {
  background-color: var(--carousel-progress-bg-dark, rgba(96, 165, 250, 0.5));
}