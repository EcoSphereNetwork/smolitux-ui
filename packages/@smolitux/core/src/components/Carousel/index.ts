// packages/@smolitux/core/src/components/Carousel/index.ts
export { default, type CarouselProps, type CarouselItem } from './Carousel';
