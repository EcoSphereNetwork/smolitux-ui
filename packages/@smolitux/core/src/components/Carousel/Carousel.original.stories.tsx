import type { Meta, StoryObj } from '@storybook/react';
import { Carousel.original } from './Carousel.original';

const meta: Meta<typeof Carousel.original> = {
  title: 'Components/Carousel.original',
  component: Carousel.original,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    children: 'Carousel.original',
  },
};

export const CustomStyle: Story = {
  args: {
    children: 'Custom Carousel.original',
    className: 'custom-style',
  },
};

export const Interactive: Story = {
  args: {
    children: 'Interactive Carousel.original',
    onClick: () => alert('Clicked!'),
  },
};
