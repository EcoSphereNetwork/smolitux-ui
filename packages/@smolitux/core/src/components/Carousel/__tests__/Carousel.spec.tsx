import React from 'react';
import { render } from '@testing-library/react';
import { Carousel } from '../Carousel';

// Mock fÃ¼r useTheme
jest.mock('@smolitux/theme', () => ({
  useTheme: () => ({
    colors: {
      primary: {
        500: '#3182ce',
      },
    },
  }),
}));

describe('Carousel Snapshots', () => {
  const mockItems = [
    { id: '1', content: <div>Slide 1</div>, ariaLabel: 'First slide' },
    { id: '2', content: <div>Slide 2</div>, ariaLabel: 'Second slide' },
    { id: '3', content: <div>Slide 3</div>, ariaLabel: 'Third slide' },
  ];

  it('renders default carousel correctly', () => {
    const { asFragment } = render(<Carousel items={mockItems} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom active index correctly', () => {
    const { asFragment } = render(<Carousel items={mockItems} defaultActiveIndex={1} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with autoPlay correctly', () => {
    const { asFragment } = render(<Carousel items={mockItems} autoPlay={3000} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom aspect ratio correctly', () => {
    const { asFragment } = render(<Carousel items={mockItems} aspectRatio="21:9" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with infinite scrolling correctly', () => {
    const { asFragment } = render(<Carousel items={mockItems} infinite />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom navigation correctly', () => {
    const { asFragment } = render(
      <Carousel 
        items={mockItems} 
        showArrows={false}
        showIndicators={false}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with custom className correctly', () => {
    const { asFragment } = render(
      <Carousel 
        items={mockItems} 
        className="custom-carousel"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});