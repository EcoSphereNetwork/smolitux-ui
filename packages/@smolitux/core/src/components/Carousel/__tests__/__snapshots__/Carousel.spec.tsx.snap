// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Snapshots renders default carousel correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 56.25%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-0%); transition: none;"
        >
          <div
            aria-hidden="false"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Vorheriges Bild"
      class="
              absolute top-1/2 left-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <button
      aria-label="Nächstes Bild"
      class="
              absolute top-1/2 right-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="absolute bottom-2 left-0 right-0 z-10 flex justify-center"
    >
      <div
        class="flex space-x-2"
      >
        <button
          aria-current="true"
          aria-label="Go to slide 1"
          class="
                w-3 h-3 rounded-full
                bg-primary-500 dark:bg-primary-400
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 2"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 3"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;

exports[`Carousel Snapshots renders with autoPlay correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 56.25%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-0%); transition: none;"
        >
          <div
            aria-hidden="false"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Vorheriges Bild"
      class="
              absolute top-1/2 left-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <button
      aria-label="Nächstes Bild"
      class="
              absolute top-1/2 right-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="absolute bottom-2 left-0 right-0 z-10 flex justify-center"
    >
      <div
        class="flex space-x-2"
      >
        <button
          aria-current="true"
          aria-label="Go to slide 1"
          class="
                w-3 h-3 rounded-full
                bg-primary-500 dark:bg-primary-400
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 2"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 3"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;

exports[`Carousel Snapshots renders with custom active index correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 56.25%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-100%); transition: none;"
        >
          <div
            aria-hidden="true"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="false"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Vorheriges Bild"
      class="
              absolute top-1/2 left-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <button
      aria-label="Nächstes Bild"
      class="
              absolute top-1/2 right-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="absolute bottom-2 left-0 right-0 z-10 flex justify-center"
    >
      <div
        class="flex space-x-2"
      >
        <button
          aria-current="false"
          aria-label="Go to slide 1"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="true"
          aria-label="Go to slide 2"
          class="
                w-3 h-3 rounded-full
                bg-primary-500 dark:bg-primary-400
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 3"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;

exports[`Carousel Snapshots renders with custom aspect ratio correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 42.857142857142854%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-0%); transition: none;"
        >
          <div
            aria-hidden="false"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Vorheriges Bild"
      class="
              absolute top-1/2 left-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <button
      aria-label="Nächstes Bild"
      class="
              absolute top-1/2 right-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="absolute bottom-2 left-0 right-0 z-10 flex justify-center"
    >
      <div
        class="flex space-x-2"
      >
        <button
          aria-current="true"
          aria-label="Go to slide 1"
          class="
                w-3 h-3 rounded-full
                bg-primary-500 dark:bg-primary-400
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 2"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 3"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;

exports[`Carousel Snapshots renders with custom className correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg custom-carousel"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 56.25%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-0%); transition: none;"
        >
          <div
            aria-hidden="false"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Vorheriges Bild"
      class="
              absolute top-1/2 left-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <button
      aria-label="Nächstes Bild"
      class="
              absolute top-1/2 right-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="absolute bottom-2 left-0 right-0 z-10 flex justify-center"
    >
      <div
        class="flex space-x-2"
      >
        <button
          aria-current="true"
          aria-label="Go to slide 1"
          class="
                w-3 h-3 rounded-full
                bg-primary-500 dark:bg-primary-400
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 2"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 3"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;

exports[`Carousel Snapshots renders with custom navigation correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 56.25%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-0%); transition: none;"
        >
          <div
            aria-hidden="false"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;

exports[`Carousel Snapshots renders with infinite scrolling correctly 1`] = `
<DocumentFragment>
  <div
    class="relative overflow-hidden rounded-lg"
  >
    <div
      class="relative w-full"
      style="padding-bottom: 56.25%;"
    >
      <div
        class="absolute inset-0"
      >
        <div
          class="flex h-full transition-transform duration-300 ease-in-out"
          style="transform: translateX(-0%); transition: none;"
        >
          <div
            aria-hidden="false"
            aria-label="First slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 1
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Second slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 2
            </div>
          </div>
          <div
            aria-hidden="true"
            aria-label="Third slide"
            class="w-full h-full flex-shrink-0"
          >
            <div>
              Slide 3
            </div>
          </div>
        </div>
      </div>
    </div>
    <button
      aria-label="Vorheriges Bild"
      class="
              absolute top-1/2 left-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M15 19l-7-7 7-7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <button
      aria-label="Nächstes Bild"
      class="
              absolute top-1/2 right-2 z-10 -translate-y-1/2
              p-2 rounded-full
              bg-white dark:bg-gray-800 bg-opacity-50 dark:bg-opacity-50
              hover:bg-opacity-75 dark:hover:bg-opacity-75
              cursor-pointer
            "
      type="button"
    >
      <svg
        class="h-6 w-6 text-gray-800 dark:text-gray-200"
        fill="none"
        stroke="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M9 5l7 7-7 7"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </button>
    <div
      class="absolute bottom-2 left-0 right-0 z-10 flex justify-center"
    >
      <div
        class="flex space-x-2"
      >
        <button
          aria-current="true"
          aria-label="Go to slide 1"
          class="
                w-3 h-3 rounded-full
                bg-primary-500 dark:bg-primary-400
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 2"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
        <button
          aria-current="false"
          aria-label="Go to slide 3"
          class="
                w-3 h-3 rounded-full
                bg-gray-300 dark:bg-gray-600 hover:bg-gray-400 dark:hover:bg-gray-500
                transition-all duration-200
                cursor-pointer
              "
          type="button"
        />
      </div>
    </div>
    <style>
      
        @keyframes fadeIn {
          from { opacity: 0; }
          to { opacity: 1; }
        }
        
        @keyframes fadeOut {
          from { opacity: 1; }
          to { opacity: 0; }
        }
      
    </style>
  </div>
</DocumentFragment>
`;
